candle-transformers/src/models/bert.rs

```diff
@@ -112,6 +112,24 @@ impl Config {
     }
 }
 
+// https://github.com/huggingface/transformers/blob/6eedfa6dd15dc1e22a55ae036f681914e5a0d9a1/src/transformers/modeling_utils.py#L877C1-L910C39
+fn get_extended_attention_mask(attention_mask: &Tensor) -> Result<Tensor> {
+    // extended_attention_mask = attention_mask[:, None, None, :]
+
+    let extended_attention_mask = attention_mask.unsqueeze(1)?.unsqueeze(2)?;
+
+    // extended_attention_mask = (1.0 - extended_attention_mask) * torch.finfo(dtype).min
+    let shape = extended_attention_mask.shape();
+
+    let on_true =
+        Tensor::new(0 as f32, extended_attention_mask.device())?.broadcast_as(shape.dims())?;
+    let on_false = Tensor::new(f32::NEG_INFINITY, extended_attention_mask.device())?
+        .broadcast_as(shape.dims())?;
+
+    let m = extended_attention_mask.where_cond(&on_true, &on_false)?;
+    Ok(m)
+}
```

Comment:
I assume this is very unoptimized? I'm not that used to candle matrix functions :/